### PR TITLE
chore: Add Fast Refresh/Hot Reloading E2E tests

### PR DESCRIPTION
**Pull request by LekoArts**
## Description

This adds a suite of tests for our Hot Reloading and the new Fast Refresh overlay.

It tests our three kinds of overlays we have:

- Compile errors
- Runtime errors (page query, static query, normal runtime error)
- GraphQL errors

It validates:

- Correct title (by `#gatsby-overlay-labelledby`
- Correct description (by `#gatsby-overlay-describedby`)
- Correct accordion title / file names
- After fixing the error the overlay goes away

## Related Issues

[ch19707]

Re-Adds tests from https://github.com/gatsbyjs/gatsby/pull/27467

### CODE REVIEW

File: e2e-tests/development-runtime/package.json [hunk 1]
```diff
@@ -33,8 +33,9 @@
   "license": "MIT",
   "scripts": {
     "build": "gatsby build",
-    "develop": "cross-env CYPRESS_SUPPORT=y ENABLE_GATSBY_REFRESH_ENDPOINT=true GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND=y gatsby develop",
+    "develop": "cross-env CYPRESS_SUPPORT=y ENABLE_GATSBY_REFRESH_ENDPOINT=true gatsby develop",
```
Comment (pieh) [inline e2e-tests/development-runtime/package.json:36]: Please re-add env var
```suggestion
    "develop": "cross-env CYPRESS_SUPPORT=y ENABLE_GATSBY_REFRESH_ENDPOINT=true GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND=y gatsby develop",
```

File: e2e-tests/development-runtime/.eslintrc [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (LekoArts) [inline e2e-tests/development-runtime/.eslintrc:1]: This was necessary to have correct GraphQL errors. With our default ESLint config we catch GraphQL errors (or at least all the ones I tested) and output it as an ESLint error (compile error) and not as GraphQL errors.

Another side-effect of this now is that we test that our default config is merged with any custom one as the tests for the ESLint checks here still pass

File: e2e-tests/development-runtime/cypress/integration/hot-reloading/error-handling/compile-error.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        `contain.text`,
+        `This error occurred during the build process and can only be dismissed by fixing the error.`
+      )
+    cy.getFastRefreshOverlay()
+      .find(`[data-gatsby-overlay="header__cause-file"] span`)
+      .should(`contain.text`, `./src/pages/error-handling/compile-error.js`)
+    cy.getFastRefreshOverlay()
+      .find(`[data-gatsby-overlay="body"] h2`)
+      .should(`contain.text`, `Source`)
+    cy.getFastRefreshOverlay().find(`[data-gatsby-overlay="body"] pre`)
```
Comment (LekoArts) [inline e2e-tests/development-runtime/cypress/integration/hot-reloading/error-handling/compile-error.js:42]: This only tests if a code preview is generated, not the contents for now

File: e2e-tests/development-runtime/cypress/integration/hot-reloading/error-handling/page-query-result-runtime-error.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+const errorPlaceholder = `false`
+const errorReplacement = `true`
+
+describe(`testing error overlay and ability to automatically recover runtime errors cause by content changes (page queries variant)`, () => {
+  it(`displays content initially (no errors yet)`, () => {
+    cy.visit(`/error-handling/page-query-result-runtime-error/`, {
+      // Hacky way to disable "uncaught:exception" message in error message itself
+      // See https://github.com/cypress-io/cypress/issues/254#issuecomment-292190924
+      onBeforeLoad: win => {
```
Comment (LekoArts) [inline e2e-tests/development-runtime/cypress/integration/hot-reloading/error-handling/page-query-result-runtime-error.js:35]: Without this change the runtime error would be a long string of "blabla Cypress caught an exception blabla then original error blabla" - so the string I want to assert has the original error but a lot of other text around it. With this the text inside `accordion__item__title` only has the error text, not more

**Review by pieh — APPROVED** [review]

**Pull request merged**